fa12_ticket_wrapper.religo

```diff
@@ -0,0 +1,89 @@
+/* 
+    To deposit FA1.2 and recieve tickets: 
+        1) Pass `Allow(...)` to the FA1.2 contract's transfer entry point, allowing this contract to transfer the desired amount of tokens.
+        2) Using the same wallet address the FA1.2 tokens were associated with, pass `Deposit(...)` to this contract's main entry point.
```

Comment:
We should also document that if Transfer fails then everything is rolled back.

Reply:
What do you think of this?

```
// To deposit FA1.2 and recieve tickets: 
// 
//   1) Pass `Allow(...)` to the FA1.2 contract's transfer entry point, allowing this contract to transfer the desired amount of tokens.
//      
//      We will use this to attempt to transfer the tokens from your address to ours. If you don't do this, the transfer will fail, and everything will be rolled back.
//
//   2) Using the same wallet address the FA1.2 tokens were associated with, pass `Deposit(...)` to this contract's main entry point.
//
//   3) A callback is taken. It will be invoked and passed the tickets corresponding to your deposit. If the callback cannot be invoked, the operation will fail, and the original transfer of tickets will be rolled back.
```